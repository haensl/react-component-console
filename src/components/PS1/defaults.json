{
  "classes": {
    "element": "Console-PS1"
  },
  "content": "$"
}
