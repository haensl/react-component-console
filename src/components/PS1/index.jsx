import React from 'react';
import './PS1.css';
import _defaults from './defaults';

export const defaults = _defaults;

const PS1 = ({
  classes = defaults.classes,
  content = defaults.content
}) => (
  <span className={ (classes && classes.element) || defaults.classes.element }>{
    content || defaults.content
  }</span>
);

export default PS1;
