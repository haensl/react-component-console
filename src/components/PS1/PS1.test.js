import React from 'react';
import renderer from 'react-test-renderer';
import PS1, { defaults } from './';

describe('PS1', () => {
  describe('empty', () => {
    let tree;

    beforeEach(() => {
      tree = renderer.create(<PS1 />)
        .toJSON();
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree.props.className).toEqual(defaults.classes.element);
    });

    it('adds the default content', () => {
      expect(tree.children.includes(defaults.content)).toBe(true);
    });
  });

  describe('content', () => {
    let tree;

    beforeEach(() => {
      tree = renderer.create(<PS1 content="~" />)
        .toJSON();
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree.props.className).toEqual(defaults.classes.element);
    });

    it('adds a tilde as content', () => {
      expect(tree.children.includes('~')).toBe(true);
    });
  });

  describe('classes', () => {
    let tree;

    beforeEach(() => {
      tree = renderer.create(<PS1 classes={{
        element: 'test-ps1'
      }} />).toJSON();
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the test-ps1 class', () => {
      expect(tree.props.className).toEqual('test-ps1');
    });

    it('adds the default content', () => {
      expect(tree.children.includes(defaults.content)).toBe(true);
    });
  });
});
