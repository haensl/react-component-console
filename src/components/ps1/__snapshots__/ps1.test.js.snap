// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PS1 classes renders as expected 1`] = `
<span
  className="test-ps1"
>
  $
</span>
`;

exports[`PS1 content renders as expected 1`] = `
<span
  className="console-ps1"
>
  ~
</span>
`;

exports[`PS1 empty renders as expected 1`] = `
<span
  className="console-ps1"
>
  $
</span>
`;
